salt/formulas.sls

```diff
@@ -51,7 +51,7 @@
     - require:
       - file: {{ basedir }}
     {%- if not update %}
-    - unless: test -e {{ gitdir }} >/dev/null 2>&1
+    - onlyif: rm -fr {{ gitdir }} >/dev/null 2>&1 | true
```

Comment:
@noelmcloughlin , this `onlyif` always evaluate to `true`. Am I missing something?

* `gitdir` exists and is removed -> `true`
* `gitdir` exists but *can't be removed for any reason* -> **or**s to `true`
```
$ sudo mkdir /tmp/deleteable
$ ls -ld /tmp/deleteable
drwxr-xr-x 2 root root 4096 Dec 13 08:28 /tmp/deleteable
$ rm -rf /tmp/deleteable; echo $?
rm: cannot remove '/tmp/deleteable': Operation not permitted
1
```
* `gitdir` does not exist, `rm -rf` won't fail -> `true`

Reply:
It is quick solution without adding new file.absent state - the goal is to clone the repo and noise from conditional is a Salt bug imho. Open to other ideas - it's an irritation.

Reply:
Yes, I understand. Comparing with other formulas, I see that we use `test -<whatever>` in many places, both with `onlyif` and with `unless`, and they don't add noise when used. 

I'm suspicious of the stderr -> stdout redirection here as the culprit of the noise, and I wonder if just using

```
unless: test -e {{ gitdir }}
```

(as I see it being used in other formulas) won't fix the issue? It's worth a try, dyt?

Reply:
When using **onlyif/unless** Salt should generate FALSE or TRUE.

However today Salt generates either TRUE or FALSE+STDERR 

Here is [example from lvm-formula](https://github.com/saltstack-formulas/lvm-formula/blob/f1f1b4e6c32ed29456a8cc6519109efb0b180b4b/lvm/pv/create.sls#L12)  - FALSE is enough - the STDERR output is unwanted.
```
[ERROR   ] Command '[u'vgdisplay', u'-c', u'cinder_volumes']' failed with return code: 5
[ERROR   ] stderr:   Volume group "cinder_volumes" not found
  Cannot process volume group cinder_volumes
```